pkg/kubelet/dockertools/docker_manager_test.go

```diff
@@ -2215,3 +2218,91 @@ func TestPruneInitContainers(t *testing.T) {
 		t.Fatal(fake.Removed)
 	}
 }
+
+func TestGetPodStatusFromNetworkPlugin(t *testing.T) {
+	const (
+		containerID      = "123"
+		infraContainerID = "9876"
+		fakePodIP        = "10.10.10.10"
+	)
+	dm, fakeDocker := newTestDockerManager()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	fnp := mock_network.NewMockNetworkPlugin(ctrl)
+	dm.networkPlugin = fnp
+
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			UID:       "12345678",
+			Name:      "foo",
+			Namespace: "new",
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{{Name: "container"}},
+		},
+	}
+
+	fakeDocker.SetFakeRunningContainers([]*FakeContainer{
+		{
+			ID:      containerID,
+			Name:    "/k8s_container_foo_new_12345678_42",
+			Running: true,
+		},
+		{
+			ID:      infraContainerID,
+			Name:    "/k8s_POD." + strconv.FormatUint(generatePodInfraContainerHash(pod), 16) + "_foo_new_12345678_42",
+			Running: true,
+		},
+	})
+
+	fnp.EXPECT().Name().Return("someNetworkPlugin")
+	fnp.EXPECT().GetPodNetworkStatus("new", "foo", kubecontainer.DockerID(infraContainerID).ContainerID()).Return(&network.PodNetworkStatus{IP: net.ParseIP(fakePodIP)}, nil)
+
+	podStatus, err := dm.GetPodStatus(pod.UID, pod.Name, pod.Namespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if podStatus.IP != fakePodIP {
+		t.Errorf("Got wrong ip, expected %v, got %v", fakePodIP, podStatus.IP)
+	}
+}
+
+func TestSyncPodGetsPodIPFromNetworkPlugin(t *testing.T) {
```

Comment:
Preventing regression is of course good :)

I am just not a big fan of merely checking function calls as it could grow really fast and become hard to maintain, while offering little value. For example, the test only checks that the function is called with arguments in a certain order, but whether they are the right order is a big question mark. We'll have to maintain both the test and the real code from now on.

I skimmed through the code, and there is no straightforward way to test this since the network plugin is completely mocked out. An integration/e2e test might be a better fit in this case. I'm inclined to remove this test, but can be convinced otherwise.